pkg/api/v20230701preview/openshiftcluster.go

```diff
@@ -133,6 +133,41 @@ const (
 	OutboundTypeLoadbalancer       OutboundType = "Loadbalancer"
 )
 
+// ResourceReference represents a reference to an Azure resource.
+type ResourceReference struct {
+	// The fully qualified Azure resource id.
+	ID string `json:"id,omitempty"`
+}
+
+// LoadbalancerProfile represents the profile of the cluster public loadbalancer.
+type LoadbalancerProfile struct {
+	// The desired managed outbound IPs for the cluster public load balancer.
+	ManagedOutboundIPs *ManagedOutboundIPs `json:"managedOutboundIps,omitempty" mutable:"true"`
+	// The list of effective outbound IP addresses of the public load balancer.
+	EffectiveOutboundIPs []EffectiveOutboundIP `json:"effectiveOutboundIps,omitempty"`
+	// The desired outbound IP resources for the cluster load balancer.
+	OutboundIPs []OutboundIP `json:"outboundIps,omitempty" mutable:"true"`
+	// The desired outbound IP Prefix resources for the cluster load balancer.
+	OutboundIPPrefixes []OutboundIPPrefix `json:"OutboundIpPrefixes,omitempty" mutable:"true"`
+	// The desired number of allocated SNAT ports per VM. Allowed values are in the range of 0 to 64000 (inclusive). The default value is 1024.
+	AllocatedOutboundPorts *int `json:"allocatedOutboundPorts,omitempty" mutable:"true"`
```

Comment:
Using a pointer here again to denote empty value.  0 value here could be ok except that ARO defaults this field to 1024, I don't know of a clean way to determine if 0 was passed in to the API or if golang just initialized it to 0.

Reply:
Perfectly fine.